pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -853,35 +827,17 @@ func (c *Controller) updateEDS(ep *v1.Endpoints, event model.Event) {
 					}
 				}
 
-				var labelMap map[string]string
-				locality, sa, uid := "", "", ""
-				if pod != nil {
-					locality = c.getPodLocality(pod)
-					sa = kube.SecureNamingSAN(pod)
-					uid = createUID(pod.Name, pod.Namespace)
-					labelMap = configKube.ConvertLabels(pod.ObjectMeta)
-				}
-
-				tlsMode := kube.PodTLSMode(pod)
+				initEndpoint := c.newIstioEndpoint(pod)
 
 				// EDS and ServiceEntry use name for service port - ADS will need to
 				// map to numbers.
 				for _, port := range ss.Ports {
-					endpoints = append(endpoints, &model.IstioEndpoint{
-						Address:         ea.IP,
-						EndpointPort:    uint32(port.Port),
-						ServicePortName: port.Name,
-						Labels:          labelMap,
-						UID:             uid,
-						ServiceAccount:  sa,
-						Network:         c.endpointNetwork(ea.IP),
-						Locality: model.Locality{
-							Label:     locality,
-							ClusterID: c.clusterID,
-						},
-						Attributes: model.ServiceAttributes{Name: ep.Name, Namespace: ep.Namespace},
-						TLSMode:    tlsMode,
-					})
+					dummySvc := &model.Service{Attributes: model.ServiceAttributes{
```

Comment:
if we are creating `dummy` things, that seems like a sign we are using the wrong abstraction?

Reply:
good, we just need ServiceAttributes here